### ISSUE TITLE
Exceptions in the plugin

### ISSUE THREAD

**Issue by singhravs** [epam/sap-commerce-intellij-idea-plugin#431]
IntelliJ IDEA version: `2023.1.2 (Ultimate Edition)`
Plugin Version: `2023.1.6`

Exception 1:

```java.lang.Throwable: Control-flow exceptions (e.g. this class com.intellij.openapi.progress.ProcessCanceledException) should never be logged. Instead, these should have been rethrown if caught.
	at com.intellij.openapi.diagnostic.Logger.ensureNotControlFlow(Logger.java:302)
	at com.intellij.idea.IdeaLogger.doLogError(IdeaLogger.java:151)
	at com.intellij.idea.IdeaLogger.error(IdeaLogger.java:142)
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:240)
	at com.intellij.openapi.vfs.newvfs.AsyncEventSupport.processEventsFromRefresh(AsyncEventSupport.java:153)
	at com.intellij.openapi.vfs.newvfs.RefreshSessionImpl.fireEventsInWriteAction(RefreshSessionImpl.java:230)
	at com.intellij.openapi.vfs.newvfs.RefreshSessionImpl.lambda$fireEvents$1(RefreshSessionImpl.java:211)
.......
......
Caused by: com.intellij.openapi.progress.ProcessCanceledException
	at com.intellij.idea.plugin.hybris.system.bean.meta.impl.BSMetaModelAccessImpl.getMetaModel(BSMetaModelAccessImpl.kt:106)
	at com.intellij.idea.plugin.hybris.vfs.listeners.BSBulkFileListener.after(BSBulkFileListener.kt:50)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeMethod(MessageBusImpl.kt:680)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeListener(MessageBusImpl.kt:640)
	at com.intellij.util.messages.impl.MessageBusImplKt.executeOrAddToQueue(MessageBusImpl.kt:482)
	at com.intellij.util.messages.impl.ToDirectChildrenMessagePublisher.publish$intellij_platform_core(CompositeMessageBus.kt:263)
	at com.intellij.util.messages.impl.MessagePublisher.invoke(MessageBusImpl.kt:442)
	at jdk.proxy1/jdk.proxy1.$Proxy73.after(Unknown Source)
	at com.intellij.openapi.vfs.newvfs.persistent.PersistentFSImpl.lambda$fireAfterEvents$18(PersistentFSImpl.java:1219)
	at com.intellij.util.Suppressions$runSuppressing$1$1.invoke(suppressions.kt:16)
	at com.intellij.util.Suppressions$runSuppressing$1$1.invoke(suppressions.kt:16)
	at com.intellij.util.Suppressions.runSuppressing(suppressions.kt:23)
	at com.intellij.util.Suppressions.runSuppressing(suppressions.kt:16)
	at com.intellij.openapi.vfs.newvfs.persistent.PersistentFSImpl.fireAfterEvents(PersistentFSImpl.java:1216)
	at com.intellij.openapi.vfs.newvfs.persistent.PersistentFSImpl.applyMultipleEvents(PersistentFSImpl.java:1195)
	at com.intellij.openapi.vfs.newvfs.persistent.PersistentFSImpl.processEventsImpl(PersistentFSImpl.java:1158)
	at com.intellij.openapi.vfs.newvfs.AsyncEventSupport.processEventsFromRefresh(AsyncEventSupport.java:150)
	... 60 more
```

Exception 2:

```java.util.NoSuchElementException: java.util.NoSuchElementException: Array contains no element matching the predicate.
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:562)
	at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:591)
	at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:672)
	at com.intellij.concurrency.JobLauncherImpl.processQueue(JobLauncherImpl.java:360)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.processInOrder(InspectionRunner.java:299)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.visitElements(InspectionRunner.java:274)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$inspect$4(InspectionRunner.java:129)
......
......
Caused by: java.util.NoSuchElementException: Array contains no element matching the predicate.
	at com.intellij.idea.plugin.hybris.impex.psi.references.FunctionTSAttributeReference$Companion.obtainTypeName(FunctionTSAttributeReference.kt:113)
	at com.intellij.idea.plugin.hybris.impex.psi.references.FunctionTSAttributeReference$Companion.findItemTypeReference(FunctionTSAttributeReference.kt:101)
	at com.intellij.idea.plugin.hybris.impex.psi.references.FunctionTSAttributeReference$Companion.access$findItemTypeReference(FunctionTSAttributeReference.kt:57)
	at com.intellij.idea.plugin.hybris.impex.psi.references.FunctionTSAttributeReference.provider$lambda$8(FunctionTSAttributeReference.kt:68)
	at com.intellij.psi.impl.PsiParameterizedCachedValue.doCompute(PsiParameterizedCachedValue.java:46)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$3(CachedValueBase.java:244)
	at com.intellij.util.CachedValueBase.computeData(CachedValueBase.java:42)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$4(CachedValueBase.java:244)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:112)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:42)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:66)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:245)
	at com.intellij.psi.impl.PsiParameterizedCachedValue.getValue(PsiParameterizedCachedValue.java:35)
	at com.intellij.psi.util.CachedValuesManager.getParameterizedCachedValue(CachedValuesManager.java:82)
	at com.intellij.idea.plugin.hybris.impex.psi.references.FunctionTSAttributeReference.multiResolve(FunctionTSAttributeReference.kt:53)
	at com.intellij.idea.plugin.hybris.codeInspection.rule.impex.ImpexUnknownTypeAttributeInspection$ImpexHeaderLineVisitor.inspect(ImpexUnknownTypeAttributeInspection.kt:58)
	at com.intellij.idea.plugin.hybris.codeInspection.rule.impex.ImpexUnknownTypeAttributeInspection$ImpexHeaderLineVisitor.visitParameter(ImpexUnknownTypeAttributeInspection.kt:44)
	at com.intellij.idea.plugin.hybris.impex.psi.impl.ImpexParameterImpl.accept(ImpexParameterImpl.java:41)
	at com.intellij.idea.plugin.hybris.impex.psi.impl.ImpexParameterImpl.accept(ImpexParameterImpl.java:46)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$processInOrder$10(InspectionRunner.java:323)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1102)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$processInOrder$11(InspectionRunner.java:331)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:229)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:187)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$processInOrder$12(InspectionRunner.java:337)
	at com.intellij.util.AstLoadingFilter.forceAllowTreeLoading(AstLoadingFilter.java:159)
	at com.intellij.util.AstLoadingFilter.forceAllowTreeLoading(AstLoadingFilter.java:151)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$processInOrder$13(InspectionRunner.java:300)
	at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:130)
	at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:119)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$processInOrder$14(InspectionRunner.java:300)
	at com.intellij.concurrency.JobLauncherImpl$1MyProcessQueueTask.lambda$call$0(JobLauncherImpl.java:299)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.concurrency.JobLauncherImpl$1MyProcessQueueTask.call(JobLauncherImpl.java:284)
	at com.intellij.concurrency.JobLauncherImpl$1MyProcessQueueTask.call(JobLauncherImpl.java:272)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedCallable.exec(ForkJoinTask.java:1428)
	... 5 more```

**Comment by mlytvyn** [comment]
can you please share:
- Plugin details
- IDEA details
- steps to reproduce (if possible)

**Comment by mlytvyn** [comment]
**Exception 1** fixed by https://github.com/epam/sap-commerce-intellij-idea-plugin/pull/412 and will be available with `2023.2.0` release

**Comment by mlytvyn** [comment]
and you're lucky today :)
**Exception 2** hopefully, should be fixed by #414.

**Comment by singhravs** [comment]
Wow! fantastic!
